addons/mail/static/src/utils/format.js

```diff
@@ -195,7 +164,7 @@ async function _generateEmojisOnHtml(htmlString) {
     for (const emoji of emojis) {
         for (const source of [...emoji.shortcodes, ...emoji.emoticons]) {
             const escapedSource = String(source).replace(/([.*+?=^!:${}()|[\]/\\])/g, "\\$1");
```

Comment:
not related to your diff: there is already a function within `@web/core/utils/strings` to escape regexps.